x-pack/plugins/encrypted_saved_objects/server/routes/key_rotation.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { RouteDefinitionParams } from '.';
+
+/**
+ * The default maximum value of from + size for searches to .kibana index. Since we cannot use scroll
+ * or search_after functionality with the .kibana index we limit maximum batch size with this value.
+ */
+const DEFAULT_MAX_RESULT_WINDOW = 10000;
+
+/**
+ * Defines routes that are used for encryption key rotation.
+ */
+export function defineKeyRotationRoutes({
+  encryptionKeyRotationService,
+  router,
+  logger,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/api/encrypted_saved_objects/rotate_key',
+      validate: {
+        query: schema.object({
+          batchSize: schema.number({
+            min: 1,
+            max: DEFAULT_MAX_RESULT_WINDOW,
+            defaultValue: DEFAULT_MAX_RESULT_WINDOW,
+          }),
+          type: schema.maybe(schema.string()),
+        }),
+      },
+      options: {
+        tags: ['access:rotateEncryptionKey'],
```

Comment:
> Is your intent for the kibana_admin role to have access to this endpoint, 

Yeah, I was thinking about this. Would you suggest any other approach for this? I remember we used/discussed this trick for the endpoints that are supposed to be used only by administrators, but I'm not sure if we now have anything that is more expressive.

Reply:
I'm personally torn between `kibana_admin` and `superuser`. In theory, it's possible that `kibana_admin` won't be authorized to access all saved object types. If one of these unauthorized types relies on encryption, then this user won't be able to rotate those objects. They would be given the impression that everything worked correctly though.

We don't have a great way of expressing this today. I think your current configuration is only available to superusers, since this access tag isn't being explicitly granted by a feature or [hard-coded for "global all"](https://github.com/elastic/kibana/blob/2e34eb239fe753eb09a36e199a8af4792cfb2434/x-pack/plugins/security/server/authorization/privileges/privileges.ts#L102).

Reply:
> I'm personally torn between kibana_admin and superuser. In theory, it's possible that kibana_admin won't be authorized to access all saved object types. If one of these unauthorized types relies on encryption, then this user won't be able to rotate those objects. They would be given the impression that everything worked correctly though.
> We don't have a great way of expressing this today. I think your current configuration is only available to superusers, since this access tag isn't being explicitly granted by a feature or hard-coded for "global all".

Ah, that's actually even better, I was mistakenly thinking that `kibana_admin` would also have access to this endpoint if it's tagged with non-registered access tag. But limiting this API to superusers only feels even safer. What do you think?

Reply:
> Ah, that's actually even better, I was mistakenly thinking that `kibana_admin` would also have access to this endpoint if it's tagged with non-registered access tag. But limiting this API to superusers only feels even safer. What do you think?

Yeah makes sense to me too ++